### PR TITLE
[feature] WoC as ChainTracker

### PR DESCRIPTION
**Pull request by sirdeggen**
https://github.com/bitcoin-sv/ts-sdk/issues/61

### CODE REVIEW

**Comment by sirdeggen** [comment]
Added default broadcaster too so that if you just do:

```javascript
await tx.broadcast()
```
it will use WoC.

Acknowledging that ARC is preferable in a production env. WoC can be useful for demos without an api key.

File: src/transaction/chaintrackers/WoChain.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { HttpClient } from "../../primitives/utils.js"
+import ChainTracker from "../ChainTracker.js"
+
+export default class WoChain implements ChainTracker {
+    URL: string
+    apiKey: string
+
+    constructor(network: string, apiKey?: string) {
+        this.URL = `https://api.whatsonchain.com/v1/bsv/${network}`
+        return this
```
Comment (ty-everett) [inline src/transaction/chaintrackers/WoChain.ts:10]:
```suggestion
```

**Review by ty-everett — CHANGES_REQUESTED** [review]
I'd generally support the addition of these new capabilities, and reluctantly set them as the defaults until options more aligned with SPV and scalability come online.

However, new tests should be introduced that exercise the code, and check any error handling behavior. You can use the `ARC.test.ts` file as a reference for testing the new (and cleverly named) WoCast broadcaster.

Extracting the HTTP client into a util was also a good move.

**Comment by sirdeggen** [comment]
@bitcoin-sv/spv-wallet gents can you please take a look at this -- specifically to align with your recent HttpClient abstraction. I think that we should align that and this PR so that there's just the one HttpClient in the library rather than multiple. @dorzepowski asking you specifically in case the team @ doesn't work.

**Comment by sirdeggen** [comment]
Maybe we should make ARC default @ty-everett - since it doesn't currently require an apiKey.

**Comment by ty-everett** [comment]
I'd rather Arc be the default, it's more future-proof than WoC and accepts EF. Can we check with TAAL and/or Gorillapool to see what their intentions are with respect to authentication over time?

**Comment by dorzepowski** [comment]
I will adjust this changes to the approach with HttpClient that is already merged in a new PR.

**Comment by dorzepowski** [comment]
New PR opened: https://github.com/bitcoin-sv/ts-sdk/pull/67
Therefore this one can be closed

**Comment by ty-everett** [comment]
Alignment completed by #67 and all merged!

**Pull request closed without merging**